src/main/datatypes/MQ2BankType.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * MacroQuest: The extension platform for EverQuest
+ * Copyright (C) 2002-2023 MacroQuest Authors
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License, version 2, as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "pch.h"
+#include "MQ2DataTypes.h"
+
+namespace mq::datatypes {
+
+
+//============================================================================
+
+enum class BankTypeMembers
+{
+	FreeSlots,
+	TotalSlots,
+	Platinum,
+	Gold,
+	Silver,
+	Copper,
+};
+
+int GetNumFreeBankSlots(int nSize)
+{
+	if (!pLocalPC || !pLocalPlayer)
+		return 0;
+
+	int freeSlots = 0;
+	for (int slot = 0; slot < GetAvailableBankSlots(); slot++)
```

Comment:
GetAvailableBankSlots() isn't super intensive or anything, but it's also not going to change:

```cpp
int availableBankSlots = GetAvailableBankSlots();
for (int slot = 0; slot < availableBankSlots; ++slot)
```